Gems/Terrain/Assets/Shaders/Terrain/TerrainPBR_ForwardPass.azsl

```diff
@@ -47,9 +45,9 @@ VSOutput TerrainPBR_MainPassVS(VertexInput IN)
     float down  = GetHeight(origUv + terrainData.m_uvStep * float2( 0.0f,  1.0f));
     float left  = GetHeight(origUv + terrainData.m_uvStep * float2(-1.0f,  0.0f));
 
-    OUT.m_bitangent = normalize(float3(0.0, terrainData.m_sampleSpacing * 2.0f, down - up));
-    OUT.m_tangent = normalize(float3(terrainData.m_sampleSpacing * 2.0f, 0.0, right - left));
-    OUT.m_normal = cross(OUT.m_tangent, OUT.m_bitangent);
+    float3 bitangent = normalize(float3(0.0, terrainData.m_sampleSpacing * 2.0f, down - up));
+    float3 tangent = normalize(float3(terrainData.m_sampleSpacing * 2.0f, 0.0, right - left));
+    OUT.m_normal = cross(tangent, bitangent);
```

Comment:
You are missing normalization if I'm not mistaken - Tangent and BiTangent are NOT ortonormal

Reply:
True. It also needs to be normalized in the pixel shader since interpolation will denormalize it again.

Reply:
Nice find @Adi-Amazon